microsoft-edge/webview2/how-to/winrt-from-js.md

```diff
@@ -345,6 +355,96 @@ Next, use the DevTools Console to demonstrate that web-side code can call the in
 
 
+<!-- =============================================== -->
+## Custom (3rd-party) WinRT components
+
+The wv2winrt tool supports custom third-party WinRT components, in addition to first-party OS WinRT APIs.
+
+![3rd-party WinRT components with wv2winrt tool](./winrt-from-js-images/wv2winrt-custom-components.png)
+
+To use custom (3rd-party) WinRT components with the wv2winrt tool, in addition to the above steps, also do the following steps:
+
+1. Add a third project (other than your main app and WinRTAdapter project) to your Visual Studio solution that implements your WinRT class.
+
+1. Have the WinRTAdapter project 'Add a reference' to your new third project containing your WinRT class.
+
+1. Update the WinRTAdapter project's Include filter in the properties to also include your new class.
+
+1. Add an additional line to `InitializeWebView2Async` to add your winrt class's namespace:
+
+   `WebView2.CoreWebView2.AddHostObjectToScript("MyCustomNamespace", dispatchAdapter.WrapNamedObject("MyCustomNamespace", dispatchAdapter));`
+
+1. For easy method calling from the web, optionally add your namespace sync proxy as a global object in script.  For example:
+
+   `window.MyCustomNamespace = chrome.webview.hostObjects.sync.MyCustomNamespace;`
+
+For an example of this, see the following branch in WebView2Samples repo:
+
+* [sample app: uwp-wv2winrt-custom-csharp-winrt](https://github.com/MicrosoftEdge/WebView2Samples/compare/uwp-wv2winrt-custom-csharp-winrt) to see an example of this.
+
+
+<!-- =============================================== -->
+## Asynchronous WinRT methods
+
+Following the steps in the above guide, you should be able to use synchronous proxies. For async method calls, you will need to use `chrome.webview.hostObjects.options.forceAsyncMethodMatches`.
+
+The [forceAsyncMethodMatches](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2#addhostobjecttoscript) property is an array of regexes, where if any regex matches a method name on a sync proxy, the method will be run asynchronously instead. Setting this to `[/Async$/]` will have it match any method ending with the suffix `Async`.  Then matching method calls work just like a method on an async proxy and returns a promise that you can await.
+
+From the above Reference documentation:
+
+| Options property | Details |
+|---|---|
+| `forceAsyncMethodMatches` | This is an array of regular expressions. When calling a method on a synchronous proxy, the method call will be performed asynchronously if the method name matches a string or regular expression in this array. Setting this value to `Async` will make any method that ends with Async be an asynchronous method call. If an async method doesn't match here and isn't forced to be asynchronous, the method will be invoked synchronously, blocking execution of the calling JavaScript and then returning the resolution of the promise, rather than returning a promise. |
```

Comment:
@david-risney @Syul968 @jasonstephen15 - keep or delete table row?  might be dup of paragraph above it.  From an Outdated comment's screenshot from Jason.

Reply:
It doesn't need to be a table since there's just one entry. It is duplicate from the table above. Since we explain a bit already on line 391 maybe just linking to the existing table above would be good rather than copying this table row down.

Reply:
deleting table & leadin now